app/graphql/resolvers/customer/customer.js

```diff
@@ -8,17 +9,31 @@ import {
 import { transformOrganisationAuthorisationToCustomerBusinessPermissionLevel } from '../../../transformers/rural-payments-portal/permissions.js'
 
 export const Customer = {
-  async business ({ customerId }, { sbi }, { dataSources }) {
+  async customerId ({ crn }, __, { dataSources }) {
+    const { id } = await dataSources.ruralPaymentsPortalApi.getCustomerByCRN(crn)
+    return id
+  },
+
+  async info ({ crn }, __, { dataSources }) {
+    const response = await dataSources.ruralPaymentsPortalApi.getCustomerByCRN(crn)
+    return ruralPaymentsPortalCustomerTransformer(response)
+  },
+
+  async business ({ crn }, { sbi }, { dataSources }) {
+    const { id } = await dataSources.ruralPaymentsPortalApi.getCustomerByCRN(crn)
     return transformPersonSummaryToCustomerAuthorisedFilteredBusiness(
-      customerId,
+      id,
       sbi,
-      await dataSources.ruralPaymentsPortalApi.getPersonSummaryByPersonId(customerId, sbi)
+      await dataSources.ruralPaymentsPortalApi.getPersonSummaryByPersonId(id, sbi)
     )
   },
-  async businesses ({ customerId }, __, { dataSources }) {
-    const summary = await dataSources.ruralPaymentsPortalApi.getPersonSummaryByPersonId(customerId)
-    return transformPersonSummaryToCustomerAuthorisedBusinesses(customerId, summary)
```

Comment:
how come you have changed customerId to crn?

Reply:
I haven't — I've changed where we're getting the data from. Before, the customerId was coming from the parent object because it was being resolved at the top-level. But now that doesn't exist there, but the CRN does. So now, rather than having the customerId passed from the parent, I'm using the crn to look-up the customerId in this resolver itself. Same in the other example.